packages/legend-application-studio/src/components/editor/side-bar/CreateNewElementModal.tsx

```diff
@@ -361,17 +366,70 @@ const NewServiceDriverEditor = observer(() => {
     } else {
       newServiceDriver.setMappingOption(val);
     }
+    //reset runtime
+    newServiceDriver.setRuntimeOption({
+      label: runtimeSelectorPlaceholder,
+      value: undefined,
+    });
+  };
+  // runtime
+  const customRuntimeLabel = '(custom)';
+  const availableRuntimes =
+    newServiceDriver.mappingOption !== undefined
+      ? editorStore.graphManagerState.usableRuntimes.filter((runtime) =>
+          runtime.runtimeValue.mappings
```

Comment:
Can we please use `getMappingCompatibleRuntimes` function here

Reply:
thanks 😊 🙏